service-base/src/main/java/org/eclipse/hono/service/AbstractProtocolAdapterBase.java

```diff
@@ -316,6 +317,26 @@ public final CommandConnection getCommandConnection() {
         return this.commandConnection;
     }
 
+    /**
+     * Sets the ResourceLimitChecks for this service.
+     *
+     * @param resourceLimitChecks The ResourceLimitChecks instance
+     */
+    @Autowired
```

Comment:
is this required? what if I do not want to enforce limits defined in a *plan*? We should probably use a default `ResourceLimitChecks` implementation that always returns `true` and then allow users to set another implementation using this method, if they want to ...

Reply:
@sophokles73 I see the point regarding enforcing limit checks. How about setting `@Autowired(required = false)`?  Before checking the limits, ensure that `ResourceLimitChecks` instance is available. If available, use it else return `Future.succeededFuture(...)`. May be it easier to have a look at the changes. I will push the changes.